model/project_parser.go

```diff
@@ -679,6 +679,21 @@ func LoadProjectInto(ctx context.Context, data []byte, opts *GetProjectOpts, ide
 			err = errors.New("trying to open include files with empty options")
 			return nil, errors.Wrapf(err, LoadProjectError)
 		}
+
+		// Validate that there are less includes than the limit.
+		settings, err := evergreen.GetConfig(ctx)
+		if err != nil {
+			return nil, errors.Wrap(err, LoadProjectError)
+		}
+		if settings == nil {
+			err = errors.New("no settings found")
+			return nil, errors.Wrap(err, LoadProjectError)
+		}
+		if settings.TaskLimits.MaxIncludesPerVersion > 0 && len(intermediateProject.Include) > settings.TaskLimits.MaxIncludesPerVersion {
```

Comment:
Can we add this as a project error rather than validating this here? https://github.com/evergreen-ci/evergreen/blob/f1ad47d558369748b427d3f79a69ffe006aead04/validator/project_validator.go#L123

Reply:
I don't think that's possible because that's validation on the project but at that stage, all the includes are already merged into the project. Includes only exist on the parser project level

Reply:
Oof okay that's a good point. Can we move this into createIntermediateProject maybe?

Reply:
Actually maybe that's okay. We could have this for loop limit to x number of files so we don't read too many files, but then we could still run project validation on the files that we _do_ read and also include an error that's like "you asked for too many files so we only gave you the first x"

Reply:
updated so that the catcher returns an error when max includes is exceeded if max includes is set